zipkin/src/main/java/zipkin2/storage/QueryRequest.java

```diff
@@ -196,10 +197,13 @@ public Builder parseAnnotationQuery(@Nullable String annotationQuery) {
       for (String ann : annotationQuery.split(" and ", 100)) {
         int idx = ann.indexOf('=');
         if (idx == -1) {
-          map.put(ann, "");
+          // put the annotation only if there is no key present already, prevents overriding more specific tags
+          map.putIfAbsent(ann.trim(), "");
```

Comment:
this is a JDK 1.8 function so failing because this type is in the core library.
```suggestion
          ann = ann.trim();
          if (!map.containsKey(ann)) map.put(ann, "");
```